cordapp-example/contracts-kotlin/src/main/kotlin/com/example/contract/CashContract.kt

```diff
@@ -32,8 +31,8 @@ class CashContract : Contract {
                 "There should be at least one input." using (cashInputs.size >= 1)
                 "There should be at least one output." using (cashOutputs.size >= 1)
 
-                val cashInSum = cashInputs.sumByLongSecure { it.value }
-                val cashOutSum = cashOutputs.sumByLongSecure { it.value }
+                val cashInSum = cashInputs.map { it.value }.fold(0L, Math::addExact)
+                val cashOutSum = cashOutputs.fold(0L) { sum, cash -> Math.addExact(sum, cash.value) }
```

Comment:
Ha! 2 ways to write the same thing. By the way, if I am not mistaken, using `{ it.value }` involves creating a new lambda that goes on the heap and will need to be garbage collected. If you want to avoid this creation you can use `.map(CashState::value)` or something like that, i.e. the method reference. So:

```kotlin
val cashInSum = cashInputs.map(CashState::value).fold(0L, Math::addExact)
```
Would be very memory efficient.

Reply:
two ways because this is all exercise ;) turns out one of them didnt work :)
as you probably see, I'm new in kotlin/java so thank you for tips about memory

Reply:
well, compiler can't accept this memory efficient version, I need to stay with lambda

Reply:
Strange. Probably wrong syntax in what I wrote.

Reply:
I wasn't able to make it fit

![image](https://user-images.githubusercontent.com/34737849/81090789-886e7880-8efe-11ea-9f6e-d3d6a5a343af.png)

Reply:
I just loaded your project in IntelliJ and this worked...

```kotlin
val cashInSum = cashInputs.map(CashState::value).fold(0L, Math::addExact)
```

Reply:
ok that's weird... :|

Reply:
Im using Java 1.8 and InteliJ 2019.3 and I'm getting compilation error when I'm using Run in InteliJ and the same when using `./gradlew test`

Reply:
Can you delete the `build`, `log` and `bin` folders?

Reply:
didn't help, I checked Kotlin plugin version, it is: 1.3.72-release-IJ2019.3-1
and my gradle is 5.4... I think there was information somewhere to use gradle 4.10, could you please tell me your versions?

Reply:
I have IntelliJ Community 2020.1, Kotlin plugin 1.3.72-release-IJ2020.1-1 and kotlin must be 1.2.71 because it is in your `constants.properties`.

Reply:
can't make this work :( ...I will move on.